src/index.ts

```diff
@@ -159,70 +163,84 @@ export class HttpClient implements IHttpClient {
       path: path,
-      options: { ...options, method: 'PUT'}
+      options: o
     });
   }
-  async patch<T>(path: string, options?: RequestOptions): Promise<T> {
+  async patch<T>(path: string, data?: Record<string, unknown>, options?: RequestOptions): Promise<T> {
+    const o = this.#processOptions( 'PATCH', data, options)
     return this.#requestJson({
       path: path,
-      options: { ...options, method: 'PATCH'}
+      options: o
     });
   }
   async delete(path: string, options?: RequestOptions): Promise<void> {
+    const o = this.#processOptions( 'DELETE', undefined, options)
     await this.#requestJson({
       path: path,
-      options: { ...options, method: 'DELETE'}
+      options: o
     });
     return undefined;
   }
-  async request(path: string, options: RequestOptions): Promise<Response<unknown>> {
+  /**
+   * @description Access to all 'got' supported options, returns the response instead of JSON.
+   * @param {string} path path or url
+   * @param {RequestOptions} options all options supported by 'got'
+   * @return {Response<unknown>}
+   */
+  async raw(path: string, options: RequestOptions): Promise<Response<unknown>> {
       return this.got(path, options)
   }
 }
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
-type HttpMethod = 'get' | 'post' | 'patch' | 'put' | 'delete';
+type HttpMethodCall = 'get' | 'post' | 'patch' | 'put' | 'delete';
 export class TestHttpClient implements IHttpClient {
   public _routePayloads: Record<string,Record<string, any>>
-  public calls: Record<HttpMethod, Record<string, any>>
+  public calls: Record<HttpMethodCall, Record<string, any>>
 
-  constructor(routePayloads: Record<string,Record<HttpMethod, unknown>>) {
+  constructor(routePayloads: Record<string,Record<HttpMethodCall, unknown>>) {
       this._routePayloads = routePayloads;
       this.calls = { get: {}, post: {}, patch: {}, put: {}, delete: {} };
   }
 
   public async get<T>(route: string): Promise<T> {
-      this._routePayloads.get[route];
-      return this._routePayloads.get[route];
+      // this._routePayloads.get[route];
+      const response = this._routePayloads.get[route]
+      return Promise.resolve(response);
   }
 
-  public async post<T>(route: string, payload: any): Promise<T> {
-      this.calls.post[route] = payload || {};
+  public async post<T>(route: string, data: any): Promise<T> {
+      this.calls.post[route] = data || {};
       return this._routePayloads.post[route];
   }
 
-  public async put<T>(route: string, payload: any): Promise<T> {
-      this.calls.put[route] = payload || {};
+  public async put<T>(route: string, data: any): Promise<T> {
+      this.calls.put[route] = data || {};
       return this._routePayloads.put[route];
   }
 
-  public async patch<T>(route: string, payload: any): Promise<T> {
-      this.calls.patch[route] = payload || {};
+  public async patch<T>(route: string, data: any): Promise<T> {
+      this.calls.patch[route] = data || {};
```

Comment:
I think I see the answer to your "why does this work even though we have an interface" question. This is because these are marked as `any`. What prevents you from using the same types as on the public interface?